### DISCUSSION TITLE
Model getters (dynamic fields, derived data, virtuals, composite)

### DISCUSSION THREAD

**Discussion by luwes** [strapi/strapi#7465] [Legacy Topics]
What's the best way to add these dynamic fields?

Simple example is if you have a column `firstname` and `lastname` and I want to output the `fullname` in the JSON response of that model each time. In my case it's for generating a signed url from a hashid which is derived from the model id.

(Preferably not using a custom controller/service and not using the `afterFind` and `afterFindOne` lifecycle hooks because they don't populate the model in every case)

edit:
basically this https://github.com/bookshelf/bookshelf/wiki/Plugin:-Virtuals
I'm using Postgres

**Comment by luwes** [comment]
Gonna answer my own question here, found out Bookshelf plugins can be installed.

```js
// config/functions/bookshelf.js

'use strict';

module.exports = (bookshelf, connection) => {
  bookshelf.plugin('bookshelf-virtuals-plugin');
};
```

then in your model put the `virtuals` object in `options`. for example:

```js
// api/post/models/post.js

exports.options = {
  virtuals: {
    url() {
      return '';
    }
  }
};
```
